### PR TITLE
New linear solver that executes a user defined solve function or method.

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**

### CODE REVIEW

File: openmdao/docs/index.rst [hunk 1]
```diff
@@ -1,7 +1,3 @@
-**********************
```
Comment (kmarsteller) [inline openmdao/docs/index.rst]: Is there a reason you took out the main document header?  Just didn't like it? :)

Comment (Kenneth-T-Moore) [reply openmdao/docs/index.rst]: There was a big merge conflict, with the new doc changes, so this must have happened during that. Good thing you checked this.

File: openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst [hunk 2]
```diff
@@ -13,4 +13,5 @@ Linear Solvers
     linear_runonce.rst
     direct_solver.rst
     petsc_ksp.rst
-    scipy_iter_solver.rst
\ No newline at end of file
+    scipy_iter_solver.rst
+    linear_user_defined.rst
```
Comment (kmarsteller) [inline openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst]: I see you adding the doc to the proper index, but I don't see the new doc in here?

File: openmdao/docs/index.rst (hunk 1 above)
Comment (kmarsteller) [reply openmdao/docs/index.rst]: Ahhh, now I get it.

File: openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst (hunk 2 above)
Comment (kmarsteller) [reply openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst]: I see the doc several commits back.  But it's gone now.  I'm guessing your doc also got removed in the merge conflicts with the new docs struct?

Comment (Kenneth-T-Moore) [reply openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst]: Not sure why that didn't get added when I did a git mv. Anyway, I've restored this missing piece.

Comment (kmarsteller) [reply openmdao/docs/feature_reference/building_blocks/solvers/linear/index.rst]: OK, that with the return of the main docs header get you my approval!

**Pull request merged**